Description: port to hurd arch
 This patch fixes libstorj build for hurd-i386.
 .
 The build failed because libuv had undefined references to symbols it
 should have defined internally.
Bug-Debian: https://bugs.debian.org/881882
Author: Samuel Thibault <sthibault@debian.org>
Origin: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=881882
Index: libuv1-1.22.0/Makefile.am
===================================================================
--- libuv1-1.22.0.orig/Makefile.am
+++ libuv1-1.22.0/Makefile.am
@@ -391,6 +391,13 @@ libuv_la_SOURCES += src/unix/bsd-ifaddrs
 test_run_tests_LDFLAGS += -lutil
 endif
 
+if HURD
+uvinclude_HEADERS += include/uv/posix.h
+libuv_la_SOURCES += src/unix/no-fsevents.c \
+                    src/unix/posix-hrtime.c \
+                    src/unix/posix-poll.c
+endif
+
 if LINUX
 uvinclude_HEADERS += include/uv/linux.h
 libuv_la_CFLAGS += -D_GNU_SOURCE
Index: libuv1-1.22.0/configure.ac
===================================================================
--- libuv1-1.22.0.orig/configure.ac
+++ libuv1-1.22.0/configure.ac
@@ -56,6 +56,7 @@ AM_CONDITIONAL([CYGWIN],   [AS_CASE([$ho
 AM_CONDITIONAL([DARWIN],   [AS_CASE([$host_os],[darwin*],       [true], [false])])
 AM_CONDITIONAL([DRAGONFLY],[AS_CASE([$host_os],[dragonfly*],    [true], [false])])
 AM_CONDITIONAL([FREEBSD],  [AS_CASE([$host_os],[*freebsd*],     [true], [false])])
+AM_CONDITIONAL([HURD],     [AS_CASE([$host_os],[gnu*],          [true], [false])])
 AM_CONDITIONAL([LINUX],    [AS_CASE([$host_os],[linux*],        [true], [false])])
 AM_CONDITIONAL([MSYS],     [AS_CASE([$host_os],[msys*],         [true], [false])])
 AM_CONDITIONAL([NETBSD],   [AS_CASE([$host_os],[netbsd*],       [true], [false])])
Index: libuv1-1.22.0/include/uv/unix.h
===================================================================
--- libuv1-1.22.0.orig/include/uv/unix.h
+++ libuv1-1.22.0/include/uv/unix.h
@@ -64,6 +64,8 @@
 # include "uv/bsd.h"
 #elif defined(__CYGWIN__) || defined(__MSYS__)
 # include "uv/posix.h"
+#elif defined(__GNU__)
+# include "uv/posix.h"
 #endif
 
 #ifndef PTHREAD_BARRIER_SERIAL_THREAD
